g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java

```diff
@@ -166,13 +167,20 @@ private static void validateLocale(HttpServletRequest request)
 
 	public static void validateSourceformat(HttpServletRequest request)
 			throws ValidationException {
-		String sourceformat = request.getParameter(APIParamName.SOURCE_FORMAT) == null ? ConstantsKeys.EMPTY_STRING
+		String sourceFormat = request.getParameter(APIParamName.SOURCE_FORMAT) == null ? ConstantsKeys.EMPTY_STRING
 				: request.getParameter(APIParamName.SOURCE_FORMAT);
-		if (StringUtils.isEmpty(sourceformat)) {
-			return;
-		}
-		if (!ConstantsKeys.SOURCE_FORMAT_LIST.contains(sourceformat.toUpperCase())) {
-			throw new ValidationException(ValidationMsg.SOURCEFORMAT_NOT_VALIDE);
+		if (!StringUtils.isEmpty(sourceFormat)) {
+			sourceFormat = sourceFormat.toUpperCase();
+			if (SourceFormatUtils.isBase64Encode(sourceFormat)){
+				sourceFormat = SourceFormatUtils.formatSourceFormatStr(sourceFormat);
+				if (!StringUtils.isEmpty(sourceFormat) && !ConstantsKeys.SOURCE_FORMAT_LIST.contains(sourceFormat)) {
+					throw new ValidationException(ValidationMsg.SOURCEFORMAT_NOT_VALIDE);
+				}
+			}else{
+				if (!ConstantsKeys.SOURCE_FORMAT_LIST.contains(sourceFormat)) {
+					throw new ValidationException(ValidationMsg.SOURCEFORMAT_NOT_VALIDE);
+				}
+			}
```

Comment:
if my above comment 'why not move "BASE64" to SOURCE_FORMAT_LIST as one of elements?' is true, i think these codes can be removed.

Reply:
we need set sourceformat to (BASE64,STRING), (BASE64,SVG) ... when use the base64 encode in clients